src/components/CustomButton.vue

```diff
@@ -0,0 +1,108 @@
+<script setup>
+import { defineProps, defineEmits } from "vue";
+const props = defineProps({
+  iconSource: {
+    type: String,
+    default: "",
+  },
+  iconAlternativeText: {
+    type: String,
+    default: "",
+  },
+  label: {
+    type: String,
+    default: "",
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+defineEmits({
+  click: {},
+});
```

Comment:
```suggestion
defineEmits(['click'])
```

See more [here](https://vuejs.org/api/sfc-script-setup#defineprops-defineemits).